ui/app/actions.js

```diff
@@ -669,12 +671,36 @@ function markAccountsFound() {
 // config
 //
 
+// default rpc target refers to localhost:8545 in this instance.
+function setDefaultRpcTarget (rpcList) {
+  log.debug(`background.setDefaultRpcTarget`)
+  background.setRpcTarget('http://localhost:8545')
+  return (dispatch) => {
+    dispatch({
+      type: actions.SET_RPC_TARGET,
+      value: 'http://localhost:8545',
+    })
+    dispatch({
+      type: actions.SET_RPC_LIST,
+      value: rpcList,
+    })
+  }
+}
+
 function setRpcTarget (newRpc) {
-  log.debug(`background.setRpcTarget`)
-  background.setRpcTarget(newRpc)
-  return {
-    type: actions.SET_RPC_TARGET,
-    value: newRpc,
+  return (dispatch) => {
+    log.debug(`background.setRpcTarget`)
+    background.setRpcTarget(newRpc)
```

Comment:
So you're still using `setRpcTarget` in both functions, so I'm still a little curious about how edge cases are handled:

- What happens if someone enters `http://localhost:8545` as a custom RPC?  Does it show both items being selected?

Maybe that's the only edge case, so maybe it's not worth worrying about, but earlier I'd proposed cleanly separating these two actions into two different methods on the `background` object: The MetaMaskController could have *both* a `setDefaultRpc()` method, *and* a `setCustomRpc()` method, which could implicitly update the frequent rpc list.